src/app/space/settings/areas/areas.component.ts

```diff
@@ -14,17 +15,22 @@ import { IModalHost } from '../../wizard/models/modal-host';
   styleUrls: ['./areas.component.less']
 })
 export class AreasComponent implements OnInit, OnDestroy {
+
+  @ViewChild('createAreaDialog') createAreaDialog: CreateAreaDialogComponent;
```

Comment:
Why does this need to be 'createAreaDialog' rather than CreateAreaDialogComponent?

Reply:
The line 

`  @ViewChild('CreateAreaDialogComponent') CreateAreaDialogComponent: CreateAreaDialogComponent;` makes the linked complain that CreateAreaDialogComponent is referenced before instantiation. `createAreaDialogComponent` with a little c is fine, i've changed that.

Reply:
I don't mean the string `'CreateDialogAreaComponent'`, I mean the type `CreateDialogAreaComponent` as the parameter for the decorator. The `@ViewChild(ModalDirective)` immediately below does this, for example.

Reply:
Ah, I misunderstood. This has been fixed.